zjit/src/hir.rs

```diff
@@ -1844,6 +1866,19 @@ impl Function {
                             // If we know that self is reasonably monomorphic from profile information, guard and use it to fold the lookup at compile-time.
                             // TODO(max): Figure out how to handle top self?
                             let Some(recv_type) = self.profiled_type_of_at(recv, frame_state.insn_idx) else {
+                                if get_option!(stats) {
+                                    match self.is_polymorphic_at(recv, frame_state.insn_idx) {
+                                        Some(true) => {
+                                            self.push_insn(block, Insn::IncrCounter(Counter::dynamic_send_polymorphic));
+                                        },
+                                        Some(false) => {
+                                            panic!("Should not have monomorphic profile at this point in this branch");
+                                        },
+                                        None => {
+                                            self.push_insn(block, Insn::IncrCounter(Counter::dynamic_send_no_profiles));
+                                        }
+                                    }
```

Comment:
I want this counter code to take up less space if possible

```suggestion
                                    match self.is_polymorphic_at(recv, frame_state.insn_idx) {
                                        Some(true) => self.push_insn(block, Insn::IncrCounter(Counter::dynamic_send_polymorphic)),
                                        Some(false) => panic!("Should not have monomorphic profile at this point in this branch"),
                                        None => self.push_insn(block, Insn::IncrCounter(Counter::dynamic_send_no_profiles)),
                                    }
```